lib/cache/cache.go

```diff
@@ -3544,3 +3545,16 @@ func (c *Cache) GetAccessGraphSettings(ctx context.Context) (*clusterconfigpb.Ac
 	}
 	return rg.reader.GetAccessGraphSettings(ctx)
 }
+
+func (c *Cache) GetProvisioningState(ctx context.Context, downstream services.DownstreamID, id services.ProvisioningStateID) (*provisioningv1.PrincipalState, error) {
+	ctx, span := c.Tracer.Start(ctx, "cache/GetProvisioningState")
+	defer span.End()
+
+	rg, err := readCollectionCache(c, c.collections.provisioningStates)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	defer rg.Release()
+
+	return rg.reader.GetProvisioningState(ctx, downstream, id)
```

Comment:
Is this going to benefit from having a read through on NotFound? We have that for some point get operations.

Reply:
This can be added later. Currently, there doesn’t seem to be a case where GetProvisioningState needs to reach the backend when NotFound. The AWS IC provisioning logic is self-healing, so if the provisioning state hasn’t yet successfully propagated to the cache, the provisioning operation will be retried.